src/utils/atoms/createResourceAtom.ts

```diff
@@ -90,6 +92,7 @@ function createResourceFetcherAtom<P, T>(
         newState.canceled = false;
         schedule(async (dispatch, ctx: ResourceCtx<P>) => {
           const version = (ctx.version ?? 0) + 1;
+          const nextParams = ctx.lastParams || null;
```

Comment:
see my comment for line 143. probably this line not needed. you assign **nextParams** to **lastParams** while regular **params** have more recent values and surely this parameter can be even more outdated after async fetch. so, by the time this parameter is needed (line 143) **ctx.lastParams** will probably have newver value so better use it directly